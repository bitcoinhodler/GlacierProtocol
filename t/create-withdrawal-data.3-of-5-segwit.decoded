{
  "txid": "03f2203c4451f6776d7511536ce357cc16ba77d51e4118ae5ed929cd3e733104",
  "hash": "4811351d396e6c840d9b70b087a401f95077b79fec2ba2fb9321f3541caf399e",
  "version": 2,
  "size": 514,
  "vsize": 219,
  "weight": 874,
  "locktime": 0,
  "vin": [
    {
      "txid": "2e861109075eeceda2fa482affb403dde93c812b86448f3f8ea96dced7412c37",
      "vout": 4,
      "scriptSig": {
        "asm": "00208e03bd637ebfec049d19df78d90085141eacbd14034793155f3822fd58d3dc5e",
        "hex": "2200208e03bd637ebfec049d19df78d90085141eacbd14034793155f3822fd58d3dc5e"
      },
      "txinwitness": [
        "",
        "304402206577e8bd38e434e7369f33752198646876d4f65dd1e2353184b76dbb3064ce3a02200561b134bce4a661fb7771efb46fb104d0c68bf433dfaef3ac8ec02df84661da01",
        "304402201214dfef3feb80005bd3af6814467090d3728747360556e23b353755f0985606022064af2c719fab928318f7c13c2768c6965738b0323d3656463dd17e16e4b1207401",
        "304402203c52dc4a245290b435b95db3d6504e22fcd60c319265506b43717e002d85310c02201706a9e06daaf42d5977ac833a73b01e6e307798f419b31379add24734338fb601",
        "53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.19997810,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 b6c7733e23a271e57ca16e5cd885174917968db9 OP_EQUALVERIFY OP_CHECKSIG",
        "desc": "addr(mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99)#fxf0ktmw",
        "hex": "76a914b6c7733e23a271e57ca16e5cd885174917968db988ac",
        "address": "mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99",
        "type": "pubkeyhash"
      }
    }
  ]
}
