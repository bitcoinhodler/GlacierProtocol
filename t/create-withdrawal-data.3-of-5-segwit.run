#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2MvdcuGhBWhwiHjTr7XUSydtv5FmMge93jS
53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155ae
mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
1
0200000001a80e97be4e3c73088d113e5a7fc871371ea491ab4b87aadad1dadc85901529a1010000006a4730440220668dbf3ec70483b05a87ac3205cd998fb36485dfceac18e68a2dfcbe771c7b9d022074de676be081d345114329826a41106c1097897053462f751cfac1c2c17d3f8f012103b31b6e234d8352672ba5b8aecfde3051109b2655528441feed27f847e80d3802ffffffff05002d31010000000017a91485f2b19245a5c3d584b1267848d9b2a7de1dc88f8790409700000000001976a91453f4e05bf61b2750b11d27fb41dd6ca11bae748b88ac002d31010000000017a914a523f8b4104ccc6abf2b9e9dbdd4fbefcac133d887002d31010000000017a9142354ab6f8b4e19a4edb8ed22efb829ce4244193f87002d31010000000017a9142524a7e29329a636bf4c1d8dea0dc6a087e5d9168700000000
3
cN5pwP2oCxokXYFdRVYczUqLqdA8GYqGWfRjaprrQXgHrS6PTHaD
cVyhd5Ei9HX2wSny8S2nYYtW2ubiUPhzdx1JUiSTjNicQJYTGLQV
cUv81BVz1JWuA9uwECPvfsoyA1TqmFJBJaQUPYrxkdf44zQSiTZY
10
y

y
INPUT
