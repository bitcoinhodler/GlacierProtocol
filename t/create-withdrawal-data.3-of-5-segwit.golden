Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.20000000 BTC

How many private keys will you be signing this transaction with (at least 3 required to complete transaction)?
#: Key #1: Key #2: Key #3: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00002510 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00002510, you have 0.19997490 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.19997810 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.20000000 BTC in unspent inputs from cold storage address 2MvdcuGhBWhwiHjTr7XUSydtv5FmMge93jS
0.19997810 BTC going to destination address mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
Fee amount: 0.00002190

Signing with private keys: 
cN5pwP2oCxokXYFdRVYczUqLqdA8GYqGWfRjaprrQXgHrS6PTHaD
cVyhd5Ei9HX2wSny8S2nYYtW2ubiUPhzdx1JUiSTjNicQJYTGLQV
cUv81BVz1JWuA9uwECPvfsoyA1TqmFJBJaQUPYrxkdf44zQSiTZY


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10 satoshis per vbyte

Raw signed transaction (hex):
02000000000101372c41d7ce6da98e3f8f44862b813ce9dd03b4ff2a48faa2edec5e070911862e04000000232200208e03bd637ebfec049d19df78d90085141eacbd14034793155f3822fd58d3dc5effffffff0172243101000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac050047304402206577e8bd38e434e7369f33752198646876d4f65dd1e2353184b76dbb3064ce3a02200561b134bce4a661fb7771efb46fb104d0c68bf433dfaef3ac8ec02df84661da0147304402201214dfef3feb80005bd3af6814467090d3728747360556e23b353755f0985606022064af2c719fab928318f7c13c2768c6965738b0323d3656463dd17e16e4b120740147304402203c52dc4a245290b435b95db3d6504e22fcd60c319265506b43717e002d85310c02201706a9e06daaf42d5977ac833a73b01e6e307798f419b31379add24734338fb601ad53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155ae00000000

Transaction fingerprint (md5):
49c9bf5e653ecf79f7bcdc552ef16610
QR code for transaction written to transaction.png
